packages/firebase_dynamic_links/README.md

```diff
@@ -71,6 +71,35 @@ final ShortDynamicLink shortenedLink = await DynamicLinkParameters.shortenUrl(
 final Uri shortUrl = shortenedLink.shortUrl;
 ```
 
+## Handle Received Dynamic Links
+
+1. In the [Firebase Console](https://console.firebase.google.com), open the Dynamic Links section.
+  - Accept the terms of service if you are prompted to do so.
+  - Take note of your project's Dynamic Links domain, which is displayed at the top of the Dynamic Links page. You need your project's Dynamic Links domain to programmatically create Dynamic Links. A Dynamic Links domain looks like `APP_CODE.app.goo.gl`.
+
+Receiving dynamic links on *iOS* requires a couple more steps than *android*. If you only want to receive dynamic links on *android*, skip to step 4. You can follow a video on the next two steps [here.](https://youtu.be/sFPo296OQqk?t=2m40s)
+
+2. In the **Info** tab of your *iOS* app's Xcode project:
+  - Create a new **URL Type** to be used for Dynamic Links.
+  - Set the **Identifier** field to a unique value and the **URL Schemes** field to be your bundle identifier, which is the default URL scheme used by Dynamic Links.
+
+3. In the **Capabilities** tab of your app's Xcode project, enable **Associated Domains** and add the following to the **Associated Domains** list:
+
+```
+applinks:APP_CODE.app.goo.gl
+```
+
+4. To receive a dynamic link, call the `retrieveDynamicLink()` method from `FirebaseDynamicLinks`:
+
+```dart
+final PendingDynamicLinkData data = await FirebaseDynamicLinks.instance.retrieveDynamicLink();
+
+final Uri deepLink = data.link;
+goToRouteUsingDeepLink(deepLink);
```

Comment:
I think this requires some additional documentation. Since we receive the pending dynamic link asynchronously, there is the question of what, if anything, to show until it arrives. If we show nothing, we may miss a frame during launch. If we show something, the view is likely to flicker as we immediately navigate to where the deep link points us (probably the poorer alternative).

The readme should provide a recommendation, e.g. to do something like
```dart
Future<void> main() async {
  final PendingDynamicLinkData data = await FirebaseDynamicLinks.instance.retrieveDynamicLink();
  final Uri deepLink = data?.link;
  runApp(widgetFor(deepLink));
}

Widget widgetFor(Uri deepLink) {
  if (deepLink == null) {
    // return app landing page
  } else {
    // calculate and return some page based on the deep link
  }
}
```